uint16 velocity
uint8 status